From ef24aae22978d4d357340074ca9a1c04d9cf81e6 Mon Sep 17 00:00:00 2001
From: Ben Brown <ben@demerara.io>
Date: Wed, 8 Jun 2022 20:40:24 +0100
Subject: [PATCH] meson: Add missing build dependencies on glu

With current versions of meson, multiple build errors occur due to
gl_wrap.h pulling in GL/glu.h, which is included by various sources,
e.g.:

    In file included from ../src/egl/opengl/xeglgears.c:45:
    ../src/util/gl_wrap.h:13:12: fatal error: GL/glu.h: No such file or directory
       13 | #  include <GL/glu.h>
          |            ^~~~~~~~~~
---
 src/egl/opengl/meson.build | 12 ++++++------
 src/util/meson.build       |  2 +-
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/egl/opengl/meson.build b/src/egl/opengl/meson.build
index ba7cadc3..f091bd17 100644
--- a/src/egl/opengl/meson.build
+++ b/src/egl/opengl/meson.build
@@ -2,15 +2,15 @@ _deps = [dep_gl, dep_m, idep_util]
 
 executable(
   'eglgears_x11', files('eglgears.c'),
-  dependencies: [_deps, idep_eglut_x11]
+  dependencies: [_deps, dep_glu, idep_eglut_x11]
 )
 executable(
   'egltri_x11', files('egltri.c'),
-  dependencies: [_deps, idep_eglut_x11]
+  dependencies: [_deps, dep_glu, idep_eglut_x11]
 )
 executable(
   'xeglgears', files('xeglgears.c'),
-  dependencies: [_deps, dep_egl, dep_x11]
+  dependencies: [_deps, dep_glu, dep_egl, dep_x11]
 )
 executable(
   'xeglthreads', files('xeglthreads.c'),
@@ -19,11 +19,11 @@ executable(
 
 executable(
   'eglgears_wayland', files('eglgears.c'),
-  dependencies: [_deps, idep_eglut_wayland]
+  dependencies: [_deps, dep_glu, idep_eglut_wayland]
 )
 executable(
   'egltri_wayland', files('egltri.c'),
-  dependencies: [_deps, idep_eglut_wayland]
+  dependencies: [_deps, dep_glu, idep_eglut_wayland]
 )
 
 executable(
@@ -39,6 +39,6 @@ executable(
 
 executable(
   'peglgears', 'peglgears.c',
-  dependencies: [dep_gl, dep_egl, dep_m, idep_util]
+  dependencies: [dep_gl, dep_glu, dep_egl, dep_m, idep_util]
 )
 
diff --git a/src/util/meson.build b/src/util/meson.build
index 9168aecf..959e2170 100644
--- a/src/util/meson.build
+++ b/src/util/meson.build
@@ -26,7 +26,7 @@ files_libutil = files(
   'trackball.c',
 )
 
-_deps = []
+_deps = [dep_glu]
 if dep_glut.found()
   files_libutil += files('shaderutil.c')
   _deps += dep_glut
-- 
2.36.1

